contracts/periphery/BeraCrocMultiSwap.sol

```diff
@@ -160,11 +159,19 @@ contract BeraCrocMultiSwap {
         return (uint128(-baseFlow), _step.base);
     }
 
-    function retire() external {
-        require(msg.sender == _deployer, "Only deployer can retire");
-        // drain the honey and stgusdc
-        IERC20Minimal(0x7EeCA4205fF31f947EdBd49195a7A88E6A91161B).transfer(_deployer, IERC20Minimal(0x7EeCA4205fF31f947EdBd49195a7A88E6A91161B).balanceOf(address(this)));
-        IERC20Minimal(0x6581e59A1C8dA66eD0D313a0d4029DcE2F746Cc5).transfer(_deployer, IERC20Minimal(0x6581e59A1C8dA66eD0D313a0d4029DcE2F746Cc5).balanceOf(address(this)));
-        selfdestruct(payable(_deployer));
+    /* @notice Refund any remaining Ether in the contract to the sender. */
+    function _refund() internal {
+        if (address(this).balance > 0) {
+            payable(msg.sender).transfer(address(this).balance);
```

Comment:
vulnerable to reentrancy attack,  msg.sender being a smart contract can call it back, did we check those cases?

Reply:
I'll switch it to use the safeTransfer, but i also don't think theres a reentrancy attack as this full yoinks whatever remaining eth exists anyways. I don't see what else there would be to take

Reply:
yea transfer should be good it protects from re-entrancy due to fixed gas.

Reply:
@calbera uniswap is pull payment(independent/seperate function to pull out money), they are using .call to send ether
tranfer is no more re-entrancy resistance post Istanbul hard-fork  but i agree with @ocnc that there shouldn't be any case re-entrancy as no funds will be left.
we should change it to use .call like uniswap and should be good then
https://consensys.io/diligence/blog/2019/09/stop-using-soliditys-transfer-now/

Reply:
ill make that change 👍